### ISSUE TITLE
Improve bson/doc macro usability

### ISSUE THREAD

**Issue by kyeah** [zonyitoo/bson-rs#62]
:wave:

After using the new serde [json!](https://github.com/serde-rs/json/blob/master/json/src/macros.rs) macro, it feels a little smarter about delineating/parsing keys and values. Improving our macro definition to allow more arbitrary values would be 👌 

```rust
doc!{ "files_id" => (self.doc.id.clone()) }
json!{ "files_id": self.doc.id.clone() }

doc!{ "success" => (code == 200) }
json!{ "success": code == 200 }

bson!(["trailing", "commas", "are", "bad"])
json!(["trailing", "commas", "are", "okay",])
```

We might also consider moving syntax over from `=>` to `:` (or allowing both of them to co-exist, if possible, for backwards-compatibility). Our initial reason for using `=>` was that macros couldn't use `:`, but this doesn't seem to be the case anymore.

**Comment by zonyitoo** [comment]
True. I want to remove that `()` around `self.doc.id.clone()` for a long time. ;)

**Comment by kyeah** [comment]
closed via #83.
